[Concurrency] Revise `Async-` related files doc

Jager-yoo opened the pull request:
- Revise `// Prints` comments style like the other stdlib files.
- Remove verbose string interpolations and extra spaces to be read more clearly.
  - (e.g.)
  ```diff
  - print ("\(error)")
  + print(error)
  ```
- Remove some unneeded parentheses.
  - (e.g.)
  ```diff
  - if (random % 5 == 0) { ... }
  + if random % 5 == 0 { ... }
  ```
- Replace the majority of ' : ' with ': '
  - (ref.) #26218
- Fix wrong indentation
- Keep files with a single empty line at the end
  - (ref.) SwiftLint Rules - ['Trailing Newline'](https://realm.github.io/SwiftLint/trailing_newline.html) and other stdlib files.

Jager-yoo committed:
[Concurrency] Revise `Async-` related files doc

- Revise '// Prints' comments style like the other stdlib files
- Remove verbose string interpolations and extra spaces
- Remove some unneeded parentheses
- Replace the majority of ' : ' with ': '
- Fix wrong indentation
- Keep files with a single empty line at the end

Jager-yoo commented:
Hi, @invalidname!
Could you please review my change proposals?
Cause most of the changes are related to your previous PR. #37383

phausler commented:
Overall the changes look sound to me. I defer to the folks doing documentation publishing for style guidance.

Jager-yoo commented:
Really appreciate it! 👍🏻

Jager-yoo commented:
Hi, @lorentey!
Could you check doc styles and ask @swift-ci to smoke test this?

invalidname commented on stdlib/public/Concurrency/AsyncStream.swift:
```diff
@@ -319,15 +319,13 @@ public struct AsyncStream<Element> {
-  ///             return Int.random(in: 1...10)
-  ///         }
-  ///         onCancel: { @Sendable () in print ("Canceled.") }
-  ///     )
```
Nice job catching this stray `)` that shouldn't be there.

Jager-yoo replied:
Thanks! 😄 
Note that I also edited the word "Canceled." → `"Cancelled."` (double `l`)

Jager-yoo replied:
@amartini51 Thanks for following up!
The [Apple Style Guide - C](https://help.apple.com/applestyleguide/#/apsgb744e4a3#apdb51b39ece6204) definitely says that we should use one `l` for `"canceled"`.

On [lines 121-122](https://github.com/apple/swift/blob/80ef80fef9a891c39a7b9a119041b5c9f0e01e21/stdlib/public/Concurrency/AsyncStream.swift#L121-L122) in the very same file, however, there is `case cancelled` in use.

```swift
/// The stream finished as a result of cancellation.
case cancelled
```

Actually, "cancelled" is being used much more than "canceled" in overall Swift repo. 🧐
Would it be better to just revert this word for now? How do you think?

amartini51 replied:
Let's revert this change please, so we're not making the problem worse.  Someone (possibly me or @invalidname) can take another pass over the file to correct the other "cancelled" spellings in a different PR.

amartini51 replied:
Regarding lines 121-121, `case cancelled` is an example of the Cocoa naming guidelines I mentioned.  You can see this style difference in API reference for `Task` and and related types — symbols with "cancelled" in its name have "canceled" in their docs.

Both style guides use the spelling "cancellation".

Jager-yoo replied:
Ohh, now I see. Documentation follows "canceled", while the API naming follows "cancelled".
Sorry for additional revert commit, and thank you so much for being part of this.

amartini51 replied:
Reading my original post, I could have been clearer that there were two style guides involved here, one for code and one for docs. Sorry about that. Thanks for the fixes in this PR!

invalidname commented on stdlib/public/Concurrency/AsyncDropFirstSequence.swift:
```diff
@@ -22,12 +22,12 @@ extension AsyncSequence {
   /// In this example, an asynchronous sequence called `Counter` produces `Int`
   /// values from `1` to `10`. The `dropFirst(_:)` method causes the modified
-  /// sequence to ignore the values `0` through `4`, and instead emit `5` through `10`:
+  /// sequence to ignore the values `1` through `3`, and instead emit `4` through `10`:
```
Right. This way matches the actual output on line 30.

invalidname reviewed (approved):
Looks good. Thanks for these corrections.

Jager-yoo commented:
Thanks for reviewing, @invalidname!
Could you ask @swift-ci to smoke test and merge this? Or should I ask someone else?

amartini51 commented on stdlib/public/Concurrency/AsyncStream.swift:
```diff
@@ -319,15 +319,13 @@ public struct AsyncStream<Element> {
-  ///     )
+  ///         await Task.sleep(1 * 1_000_000_000)
+  ///         return Int.random(in: 1...10)
+  ///     } onCancel: { @Sendable () in print("Cancelled.") }
```
```suggestion
  ///     } onCancel: { @Sendable () in print("Canceled.") }
```

amartini51 committed:
Back out spelling change.

Documentation follows Apple Style Guide spelling:

https://help.apple.com/applestyleguide/#/apsgb744e4a3#apdb51b39ece6204

amartini51 commented:
@swift-ci Please smoke test.
